lib/validators/specResolver.js

```diff
@@ -488,6 +501,281 @@ class SpecResolver {
+    //   {
+    //     "$ref": "#/definitions/Activity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/ControlActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/ExecutePipelineActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/IfConditionActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/ForEachActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/WaitActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/UntilActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/ExecutionActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/CopyActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/HDInsightHiveActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/HDInsightPigActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/HDInsightMapReduceActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/HDInsightStreamingActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/HDInsightSparkActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/CustomActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/SqlServerStoredProcedureActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/LookupActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/WebActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/GetMetadataActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/AzureMLBatchExecutionActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/AzureMLUpdateResourceActivity"
+    //   },
+    //   {
+    //     "$ref": "#/definitions/DataLakeAnalyticsUSQLActivity"
+    //   }
+    // ];
+
+    console.log("Hello");
+    return Promise.resolve(self);
+  }
+
+  createPolymorphicTree(name, discriminator, subTreeMap) {
+    if (name === null || name === undefined || typeof name.valueOf() !== 'string' || !name.trim().length) {
+      throw new Error('name is a required property of type string and it cannot be an empty string.')
+    }
+
+    if (discriminator === null || discriminator === undefined || typeof discriminator.valueOf() !== 'string' || !discriminator.trim().length) {
+      throw new Error('discriminator is a required property of type string and it cannot be an empty string.')
+    }
+
+    if (subTreeMap === null || subTreeMap === undefined || !(subTreeMap instanceof Map)) {
+      throw new Error('subTreeMap is a required property of type Map.')
+    }
+
+    let root = new PolymorphicTree(name);
+    let definitions = this.specInJson.definitions;
+
+    // Adding the model name or it's discriminator value as an enum constraint on property marked as discriminator
+    if (definitions[name] && definitions[name].properties && definitions[name].properties[discriminator]) {
```

Comment:
// Adding the model name or it's discriminator value as an enum constraint **with one value (constant)** on property marked as discriminator